test/workerinfo_test.js

```diff
@@ -1,49 +1,120 @@
-    let workerInfo = await helper.load('workerInfo', helper.loadOptions);
+    const workerInfo = await helper.load('workerInfo', helper.loadOptions);
+
     await Promise.all([
       workerInfo.provisionerSeen('prov2'),
       workerInfo.provisionerSeen('prov2'),
     ]);
     await workerInfo.provisionerSeen('prov2');
-    let result = await helper.queue.listProvisioners();
+
+    const result = await helper.queue.listProvisioners();
     assert(result.provisioners.length === 1, 'expected a provisioner');
   });
 
-  test('expiration works', async () => {
-    let Provisioner = await helper.load('Provisioner', helper.loadOptions);
-    await Provisioner.create({provisionerId: 'prov1', expires: new Date('1017-07-29')});
+  test('provisioner expiration works', async () => {
+    const Provisioner = await helper.load('Provisioner', helper.loadOptions);
 
+    await Provisioner.create({provisionerId: 'prov1', expires: new Date('1017-07-29')});
     await helper.expireWorkerInfo();
 
-    let result = await helper.queue.listProvisioners();
+    const result = await helper.queue.listProvisioners();
     assert(result.provisioners.length === 0, 'expected no provisioners');
   });
+
+  test('queue.listWorkerTypes returns an empty list', async () => {
+    const WorkerType = await helper.load('WorkerType', helper.loadOptions);
+    const result = await helper.queue.listWorkerTypes('no-provisioner');
+
+    assert(result.workerTypes.length === 0, 'did not expect any worker-types');
+  });
+
+  test('queue.listWorkerTypes returns workerTypes', async () => {
+    const WorkerType = await helper.load('WorkerType', helper.loadOptions);
+    const workerType = slugid.v4();
+
+    await WorkerType.create({provisionerId: 'prov-A', workerType, expires: new Date('3017-07-29')});
+
+    const result = await helper.queue.listWorkerTypes('prov-A');
+
+    assert(result.workerTypes.length === 1, 'expected workerTypes');
+    assert(result.workerTypes[0] === workerType, `expected ${workerType}`);
+  });
+
+  test('list worker-types (limit and continuationToken)', async () => {
+    const WorkerType = await helper.load('WorkerType', helper.loadOptions);
+    const expires = new Date('3017-07-29');
+
+    await WorkerType.create({provisionerId: 'prov1', workerType: slugid.v4(), expires});
+    await WorkerType.create({provisionerId: 'prov1', workerType: slugid.v4(), expires});
+
+    let result = await helper.queue.listWorkerTypes('prov1', {limit: 1});
+
+    assert(result.continuationToken);
+    assert(result.workerTypes.length === 1);
+
+    result = await helper.queue.listWorkerTypes('prov1', {
+      limit: 1,
+      continuationToken: result.continuationToken,
+    });
+
+    assert(!result.continuationToken);
+    assert(result.workerTypes.length === 1);
+  });
+
+  test('workerTypeSeen creates and updates a worker-type', async () => {
+    let workerInfo = await helper.load('workerInfo', helper.loadOptions);
+    let workerTypeA = slugid.v4();
```

Comment:
Ah I see. In that case, I will fix this test file to have better worker-type names.